src/main/java/uk/gov/hmcts/reform/mi/miextractionservice/component/impl/payment/PaymentFilterComponentImpl.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.reform.mi.miextractionservice.component.impl.payment;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import uk.gov.hmcts.reform.mi.miextractionservice.component.DataParserComponent;
+import uk.gov.hmcts.reform.mi.miextractionservice.component.FilterComponent;
+import uk.gov.hmcts.reform.mi.miextractionservice.domain.SourceEnum;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.time.format.DateTimeFormatter.ISO_DATE_TIME;
+
+@AllArgsConstructor
+@Component
+@Qualifier("payment")
+public class PaymentFilterComponentImpl implements FilterComponent {
```

Comment:
Can this component be made generic? 
the getDateFieldName can use a configuration property to retrieve the field.
In this case we won't need to do code changes to add new source, or at least not need new FilterComponent and just add a new configuration